sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -93,6 +119,18 @@ TEST(Context, IsCancelled)
   EXPECT_TRUE(c2.IsCancelled());
 }
 
+TEST(Context, ThrowIfCancelled)
```

Comment:
We should also add two more tests that make sure the correctness of the tree traversal both for `CancelWhen()` and the `HasKey()`/`[]operator` get value.

Right now, we just have basic tests on a single context, we'd want to get a value for a key from a context's parent, and  validate that we do look at the parent's values/cancelation time and do NOT look at the children's values/cancelation time. Having two traversal-based tests would cover those.

Reply:
CancelWhen() is private now as it was not used by any current scenarios.

Reply:
The feedback still holds for `HeyKey()` and `operator[]`. Those also do tree traversal.